File: Java/RecyclerView11BaseClassForActivityAdapter/app/src/main/java/dev/ogabek/recyclerview11_baseclassforactivityadapter/activity/BaseActivity.java
package dev.ogabek.recyclerview11_baseclassforactivityadapter.activity;

import androidx.appcompat.app.AppCompatActivity;

import java.util.ArrayList;
import java.util.List;

import dev.ogabek.recyclerview11_baseclassforactivityadapter.model.Member;
import dev.ogabek.recyclerview11_baseclassforactivityadapter.network.BaseServer;

public class BaseActivity extends AppCompatActivity implements BaseServer {

    protected String getServer() {
        if (isTester) {
            return SERVER_DEVELOPMENT;
        } else {
            return SERVER_DEPLOYMENT;
        }
    }

    protected List<Member> prepareMemberList() {
        List<Member> members = new ArrayList<>();
        for (int i = 0; i < 25; i++) {
            members.add(new Member("Ogabek", "Matyakubov"));
        }
        return members;
    }

}

File: Kotlin/RecyclerView11BaseClassForActivityAndAdapter/app/src/main/java/dev/ogabek/recyclerview_11_baseclassforactivityandadapter/activity/BaseActivity.kt
package dev.ogabek.recyclerview_11_baseclassforactivityandadapter.activity

import androidx.appcompat.app.AppCompatActivity
import dev.ogabek.recyclerview_11_baseclassforactivityandadapter.model.Member
import dev.ogabek.recyclerview_11_baseclassforactivityandadapter.network.BaseServer

open class BaseActivity: AppCompatActivity(), BaseServer {

    protected fun getServer(): String {
        return if (BaseServer.IS_TESTER) {
            BaseServer.SERVER_DEVELOPMENT
        } else {
            BaseServer.SERVER_DEPLOYMENT
        }
    }

    protected fun prepareMemberList(): List<Member> {
        val members = ArrayList<Member>()
        for (i in 0..50) members.add(Member("Ogabek Matyakubov", "@OgabekDev"))
        return members
    }

}